frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -605,6 +605,12 @@ function VirtualKeyPopup:init()
                 -- Support any function as a callback.
                 if v_func then
                     virtual_key.callback = v_func
+                else
+                    local cb = virtual_key.callback
+                    virtual_key.callback = function ()
+                        if cb then cb() end
+                        UIManager:close(self)
```

Comment:
Maybe instead of creating yet another anonymous function (althought we do so many :), you could just add` virtual_key.close_after_callback_widget = self` and do `if self.close_after_callback_widget then UIManager:close(self.close_after_callback_widget)` in the original callback

Reply:
Done.